lib/src/bootloader.rs

```diff
@@ -26,3 +26,94 @@ pub(crate) fn install_via_bootupd(
         .verbose()
         .run()
 }
+
+#[context("Installing bootloader using zipl")]
+pub(crate) fn install_via_zipl(device: &Utf8Path, boot_uuid: &str) -> Result<()> {
+    // Identify the target boot partition from UUID
+    let fs = crate::mount::inspect_filesystem_by_uuid(boot_uuid)?;
+    let boot_dir = Utf8Path::new(&fs.target);
+    let maj_min = fs.maj_min;
+
+    // Ensure that the found partition is a part of the target device
+    crate::blockdev::list_dev(device)?
+        .children
+        .with_context(|| format!("no partition found on {device:?}"))?
+        .iter()
+        .find(|dev| dev.maj_min == Some(maj_min.clone()))
+        .with_context(|| format!("partition device {:?} is not on {:?}", maj_min, device))?;
+
+    // Get an offset of the target partition by reading /sys/dev/block/<major:minor>/start
+    let sysfs_start_path = Utf8Path::new("/sys/dev/block").join(&maj_min).join("start");
+    let boot_offset = match std::fs::read_to_string(sysfs_start_path.clone()) {
+        Ok(v) => v.trim().parse::<i64>().map_err(|e| anyhow!(e)),
+        Err(e) if e.kind() == std::io::ErrorKind::NotFound => Ok(0), // partition-less disk
```

Comment:
Fine as is, but - if you want (and again this is optional) we have a nice handy helper for this in https://docs.rs/cap-std-ext/latest/cap_std_ext/dirext/trait.CapStdExtDirExt.html#tymethod.open_optional

(Using this would require opening the rootfs as a `Dir` or accepting it as an option, see e.g. https://github.com/containers/bootc/pull/649/commits/ab1149d6cbaa94dbbed24b2747a10ae5d34b350b )

Reply:
Switch to use `backfill_start`